File: branches/TPL-Dual-Solution/ASCOM.Utilities/ASCOM Diagnostics/SerialForm.vb
Imports ASCOM.Utilities
Public Class SerialForm
    Private Sub SerialForm_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        Dim SerPort As New ASCOM.Utilities.Serial, Ports() As String
        Ports = SerPort.AvailableCOMPorts
        SerPort.Dispose()
        SerPort = Nothing
        For Each Port As String In Ports
            lstSerialASCOM.Items.Add(Port)
        Next

    End Sub

    Private Sub btnSerialExit_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSerialExit.Click
        Me.Visible = False
        Me.Close()
    End Sub
End Class

File: Diagnostics/ASCOM Diagnostics/SerialForm.cs
using System;
using Microsoft.VisualBasic;

namespace ASCOM.Utilities
{
    public partial class SerialForm
    {
        /// <summary>
        /// Serial form initialiser
        /// </summary>
        public SerialForm()
        {
            InitializeComponent();
        }
        private void SerialForm_Load(object sender, EventArgs e)
        {
            var SerPort = new Serial();
            string[] Ports;
            Ports = SerPort.AvailableCOMPorts;
            SerPort.Dispose();
            foreach (string Port in Ports)
                lstSerialASCOM.Items.Add(Interaction.IIf(string.IsNullOrEmpty(Port), "Bad value - Null or empty COM port name!", Port));

        }

        private void btnSerialExit_Click(object sender, EventArgs e)
        {
            Visible = false;
            Close();
        }
    }
}